### PR TITLE
Bug 1381604 - soundcloud.com does not show up in Spotlight search on …

### PR DESCRIPTION
**Pull request by justindarc**
…iPhone

### CODE REVIEW

File: Client/Assets/Favicons.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ Object.defineProperty(window.__firefox__, 'favicons', {
       var favicons = {};
     
       for (var selector in selectors) {
-        var icons = document.head.querySelectorAll(selector);
+        var icons = document.querySelectorAll(selector);
```
Comment (farhanpatel) [inline Client/Assets/Favicons.js]: This will be more expensive if we have to query the entire document.

**Comment by farhanpatel** [comment]
About Spotlighthelper, The last I remember querying the entire body was causing redraws which could be hurting the performance of the page. You can run safari web inspector to see this.
 Does passing the entire body even make sense? What if we just passed title/description from the document head

**Comment by justindarc** [comment]
Generally, more-specific CSS selectors are actually more expensive than less-specific ones because CSS selectors work from right-to-left. As far as SpotlightHelper goes, its not actually fetching the entire body there either. Its just finding the `<title>` tag without specifying that it must also be within `<head>`. Also, the code doesn’t execute until the ‘load’ event fires too, so the `<body>` is already available at that point when this runs.

Its a very common misconception that CSS works left-to-right. This is why the `foo < bar` selector has never existed and probably never will.

https://csswizardry.com/2011/09/writing-efficient-css-selectors/

**Comment by garvankeeley** [comment]
https://www.w3.org/2005/10/howto-favicon

sites  often ignore w3c guidelines, but it appears to specify within HEAD tags

**Comment by justindarc** [comment]
Yeah. Also, I'm pretty sure the `<title>` tag is supposed to go in `<head>` according to the standards... but I digress :-)

**Pull request merged**